src/lib/utils/redisCache.ts

```diff
@@ -0,0 +1,33 @@
+/* eslint-disable no-console */
+import { DataCache } from 'next-drupal'
+import {
+  createClient,
+  RedisClientType,
+  RedisFunctions,
+  RedisModules,
+  RedisScripts,
+} from 'redis'
+
+export async function createRedisClient(
+  url
+): Promise<RedisClientType<RedisModules, RedisFunctions, RedisScripts>> {
+  return await createClient({ url })
+    .on('error', (err) => console.error(`Redis Error: ${err}`))
+    .on('connect', () => console.info('Redis connected'))
+    .on('reconnecting', () => console.info('Redis reconnecting'))
+    .on('ready', () => console.info('Redis ready!'))
+    .connect()
+}
+
+export function redisCache(
+  client: Promise<RedisClientType<RedisModules, RedisFunctions, RedisScripts>>
+): DataCache {
+  return {
+    async set(key, value) {
```

Comment:
Could we set a default TTL here and a mechanism for override via flag/option? 

Very much could be follow-on work.

Reply:
yes, we could set a default TTL. if we are only caching at build-time and doing a teardown of redis after, it may not be necessary, though.

it would be something like:
```ts
    async set(key, value) {
      return (await client).get(key, { 
        EX: option.REDIS_EXPIRE_TIME 
      })
    },
```